src/Libs/Gtk-4.0/Public/FileDialog.cs

```diff
@@ -107,4 +141,38 @@ public partial class FileDialog
 
         return tcs.Task;
     }
+
+    //TODO: Async methods should be generated automatically (https://github.com/gircore/gir.core/issues/893)
+    public Task<Gio.ListModel?> SelectMultipleFoldersAsync(Window parent)
+    {
+        var tcs = new TaskCompletionSource<Gio.ListModel?>();
+
+        var callbackHandler = new Gio.Internal.AsyncReadyCallbackAsyncHandler((sourceObject, res, data) =>
+        {
+            if (sourceObject is null)
+            {
+                tcs.SetException(new Exception("Missing source object"));
+                return;
+            }
+
+            var listValue = Internal.FileDialog.SelectMultipleFoldersFinish(sourceObject.Handle, res.Handle, out var error);
+
+            if (!error.IsInvalid)
+                tcs.SetException(new GLib.GException(error));
+            else if (listValue == IntPtr.Zero)
+                tcs.SetResult(null);
+            else
+                tcs.SetResult(new Gio.ListModelHelper(listValue, true));
```

Comment:
You should not create a new `ListModel` on each call but go through `GObject.Internal.ObjectWrapper.WrapInterfaceHandle<Gio.ListModelHelper>` or `WrapNullableInterfaceHandle<Gio.ListModelHelper>` this ensures that a list model instance is reused if it is already known by the runtime.

In your specific case you should use the nullable variant as the pointer is nullable. In this case you can remove the `else if (listValue == IntPtr.Zero)` block as it will be handled for you automatically.